packages/compass-crud/src/components/bulk-update-dialog.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useMemo, useState } from 'react';
+import Document from './document';
+import HadronDocument from 'hadron-document';
+
+import { toJSString } from 'mongodb-query-parser';
+
+import {
+  FormModal,
+  css,
+  cx,
+  spacing,
+  palette,
+  Label,
+  Banner,
+  BannerVariant,
+  KeylineCard,
+  Description,
+  Link,
+  InfoSprinkle,
+  useDarkMode,
+  TextInput,
+} from '@mongodb-js/compass-components';
+
+import type { Annotation } from '@mongodb-js/compass-editor';
+import { CodemirrorMultilineEditor } from '@mongodb-js/compass-editor';
+
+import type { BSONObject } from '../stores/crud-store';
+import type { UpdatePreview } from 'mongodb-data-service';
+
+const columnsStyles = css({
+  marginTop: spacing[4],
+  display: 'grid',
+  width: '100%',
+  gap: spacing[4],
+  gridTemplateColumns: 'repeat(2, minmax(0, 1fr))',
+});
+
+const queryStyles = css({
+  display: 'flex',
+  gap: spacing[4],
+  flexDirection: 'column',
+});
+
+const queryFieldStyles = css({});
+
+const updateFieldStyles = css({
+  flex: 1,
+});
+
+const descriptionStyles = css({
+  marginBottom: spacing[2],
+});
+
+const previewStyles = css({
+  contain: 'size',
+  overflow: 'scroll',
+});
+
+const previewDescriptionStyles = css({
+  display: 'inline',
+});
+
+const codeContainerStyles = css({
+  paddingTop: spacing[2],
+  paddingBottom: spacing[2],
+});
+
+const codeDarkContainerStyles = css({
+  background: palette.gray.dark4,
+});
+
+const codeLightContainerStyles = css({
+  backgroundColor: palette.gray.light3,
```

Comment:
Not sure how this correlates with the styles below: this is exactly the same color we use in the editor? Is it only the dark mode that's different? Why?

Reply:
To match the design. Otherwise it is white. We do the exact same thing in the aggregation stage editor: https://github.com/mongodb-js/compass/blob/dbf1b0d71c1efcb870e4690e91bac37fc3f1e6db/packages/compass-aggregations/src/components/stage-editor/stage-editor.tsx#L43

There's a ticket to refactor that so we don't keep duplicating it along with the banner area at the bottom.

Reply:
Not sure I follow, default multiline background is the same gray color: https://github.com/mongodb-js/compass/blob/c3a174ea2272f21df453d929f0542fd33e675284/packages/compass-editor/src/json-editor.tsx#L141

Sounds good if we're planning to do a follow-up clean up for this 👍